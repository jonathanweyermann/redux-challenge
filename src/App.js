import React from 'react';
import {
	Router,
	Route,
	Switch,
	Link
} from 'react-router-dom';
import Machines from './Machines';
import Machine from './Machine';
import WebsocketLoader from './WebsocketLoader';
import './App.css';
import ReduxThunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from './reducers';

import PropTypes from 'prop-types';

const store = createStore(rootReducer, {}, applyMiddleware(ReduxThunk));
import 'bootstrap/dist/css/bootstrap.min.css';
import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

function App (props) {

	return (
		<Provider store={store}>
			<WebsocketLoader />
			<Router history={history} >
				<div className='App'>
					<header className='App-header'>
						<img alt='logo' height='272' width='800' src='https://i.imgur.com/jcvsFKh.png' />
					</header>

					<nav className='App-nav'>
						<Link to='/'>Home</Link>
						<Link to='/machines'>Machines</Link>
					</nav>

					<Switch>
						<Route exact path='/machines'>
							<Machines history={history} />
						</Route>
						<Route path='/machines/:id'>
							<Machine history={history} />
						</Route>
					</Switch>
				</div>
			</Router>
		</Provider>
	);
}

App.propTypes = {
	getMachines: PropTypes.array
};


export default App;
