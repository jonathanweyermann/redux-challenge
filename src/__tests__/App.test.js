import renderer from 'react-test-renderer';

import React from 'react';
import { mount } from 'enzyme';
import App from '../App';

import mock from 'xhr-mock';
import { render } from '@testing-library/react';

describe('Main page', () => {
  it('shows a blank page except for some links', () => {
		const wrapper = mount(<App />);
    expect(wrapper).toMatchSnapshot();
  });

	it('renders /machines link', () => {
	  const { getByText } = render(<App />);
	  const linkElement = getByText(/machines/i);
	  expect(linkElement).toBeInTheDocument();
	});
});
