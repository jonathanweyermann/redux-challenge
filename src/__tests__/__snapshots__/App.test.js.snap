// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page shows a blank page except for some links 1`] = `
<App>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(WebsocketLoader)>
      <WebsocketLoader
        getMachines={[Function]}
        machines={
          Object {
            "machines": Array [],
            "name": "",
          }
        }
        updateMachinesLocally={[Function]}
      />
    </Connect(WebsocketLoader)>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div
        className="App"
      >
        <header
          className="App-header"
        >
          <img
            alt="logo"
            height="272"
            src="https://i.imgur.com/jcvsFKh.png"
            width="800"
          />
        </header>
        <nav
          className="App-nav"
        >
          <Link
            to="/"
          >
            <LinkAnchor
              href="/"
              navigate={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </LinkAnchor>
          </Link>
          <Link
            to="/machines"
          >
            <LinkAnchor
              href="/machines"
              navigate={[Function]}
            >
              <a
                href="/machines"
                onClick={[Function]}
              >
                Machines
              </a>
            </LinkAnchor>
          </Link>
        </nav>
        <Switch />
      </div>
    </Router>
  </Provider>
</App>
`;
